common/scala/src/main/scala/whisk/core/connector/MessagingProvider.scala

```diff
@@ -34,6 +34,5 @@ trait MessagingProvider extends Spi {
                   maxPeek: Int = Int.MaxValue,
                   maxPollInterval: FiniteDuration = 5.minutes)(implicit logging: Logging): MessageConsumer
   def getProducer(config: WhiskConfig, ec: ExecutionContext)(implicit logging: Logging): MessageProducer
-  def ensureTopic(config: WhiskConfig, topic: String, topicConfig: Map[String, String])(
-    implicit logging: Logging): Boolean
+  def ensureTopic(config: WhiskConfig, topic: String, topicConfig: String)(implicit logging: Logging): Boolean
```

Comment:
Now, `WhiskConfig` can be removed, right?

Reply:
No.  We need it to get to kafkaHosts (same as the other two SPI methods).  https://github.com/apache/incubator-openwhisk/blob/d69f8db02be70db0da1e07614124ea2e08dd6826/common/scala/src/main/scala/whisk/connector/kafka/KafkaMessagingProvider.scala#L52